dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java

```diff
@@ -0,0 +1,227 @@
+            for (int i = 0; i < elements.length; i++) {
+                if (values.length <= i) {
+                    elements[i].appendChild(doc.createTextNode(""));
+                } else {
+                    elements[i].appendChild(doc.createTextNode(values[i]));
+                }
+                el.appendChild(elements[i]);
+            }
+
+            return doc;
+        } catch (ParserConfigurationException e) {
+            return null;
+        }
+    }
+
+    public static Node getSize(String mdValue) {
+        String ns = "http://www.clarin.eu/cmd/";
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setNamespaceAware(true);
+        DocumentBuilder builder;
+        try {
+            builder = factory.newDocumentBuilder();
+            Document doc = builder.newDocument();
+            Element el = doc.createElementNS(ns, "size");
+            doc.appendChild(el);
+            Element size = doc.createElementNS(ns, "size");
+            Element unit = doc.createElementNS(ns, "unit");
+
+            String[] values = mdValue
+                    .split(DCInput.ComplexDefinitions.getSeparator(), -1);
+
+            Element[] elements = {size, unit};
+            for (int i = 0; i < elements.length; i++) {
+                if (values.length <= i) {
+                    elements[i].appendChild(doc.createTextNode(""));
+                } else {
+                    elements[i].appendChild(doc.createTextNode(values[i]));
+                }
+                el.appendChild(elements[i]);
+            }
+            return doc;
+        } catch (ParserConfigurationException e) {
+            return null;
+        }
+    }
+
+    public static Node getAuthor(String mdValue) {
+        String ns = "http://www.clarin.eu/cmd/";
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setNamespaceAware(true);
+        DocumentBuilder builder;
+        try {
+            builder = factory.newDocumentBuilder();
+            Document doc = builder.newDocument();
+            Element el = doc.createElementNS(ns, "author");
+            doc.appendChild(el);
+            Element last = doc.createElementNS(ns, "lastName");
+
+            String[] values = mdValue
+                    .split(",", 2);
+
+            last.appendChild(doc.createTextNode(values[0]));
+            el.appendChild(last);
+            if (values.length > 1) {
+                Element first = doc.createElementNS(ns, "firstName");
+                first.appendChild(doc.createTextNode(values[1]));
+                el.appendChild(first);
+            }
+            return doc;
+        } catch (ParserConfigurationException e) {
+            return null;
+        }
+    }
+
+    public static boolean hasOwnMetadata(List<MetadataValue> metadataValues) {
+        if (metadataValues.size() == 1 && metadataValues.get(0).getValue().equalsIgnoreCase("true")) {
```

Comment:
that is different from == 1.
I don't know why they did it like this. Maybe it cannot have more data than one, then it would be ambiguous.
Taken from here https://github.com/ufal/clarin-dspace/blob/8780782ce2977d304f2390b745a98eaea00b8255/dspace-api/src/main/java/org/dspace/content/Item.java#L2219